internal/resolution/resolver_test.go

```diff
@@ -19,112 +18,115 @@ import (
+		&v1alpha1.Operator{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: "prometheus",
 			},
 			Spec: v1alpha1.OperatorSpec{
 				PackageName: "prometheus",
 			},
-		}
-		client := FakeClient(resource)
-		entitySource := FailEntitySource{}
-		resolver := resolution.NewOperatorResolver(client, entitySource)
-		solution, err := resolver.Resolve(context.Background())
-		Expect(solution).To(BeNil())
-		Expect(err).To(HaveOccurred())
-	})
-
-	It("should return an error if the client throws an error", func() {
-		client := NewFailClientWithError(fmt.Errorf("something bad happened"))
-		entitySource := input.NewCacheQuerier(testEntityCache)
-		resolver := resolution.NewOperatorResolver(client, entitySource)
-		solution, err := resolver.Resolve(context.Background())
-		Expect(solution).To(BeNil())
-		Expect(err).To(HaveOccurred())
-	})
-})
+		},
+		&v1alpha1.Operator{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "packageA",
+			},
+			Spec: v1alpha1.OperatorSpec{
+				PackageName: "packageA",
+			},
+		},
+	}
+	
+	errorClient := NewFailClientWithError(fmt.Errorf("something bad happened"))
+
+	for _, tt := range []struct {
+		Name				string
+		Client				client.Client
+		EntitySource		input.EntitySource
+		SelectedVariableCnt	int
+		ExpectedError		error
+	}{	
+		{
+			Name: "should resolve the packages described by the available Operator resources",
+			Client: FakeClient(testResource...),
+			EntitySource: testEntitySource,
+			SelectedVariableCnt: 4,
+			ExpectedError: nil,
+		},
+		{
+			Name: "should not return an error if there are no Operator resources",
+			Client: FakeClient([]client.Object{}...),
+			EntitySource: testEntitySource,
+			SelectedVariableCnt: 0,
+			ExpectedError: nil,
+		},
+		{
+			Name: "should return an error if the entity source throws an error",
+			Client: FakeClient(testResource...),
+			EntitySource: FailEntitySource{},
+			ExpectedError: fmt.Errorf("error calling filter in entity source"),
+		}, 
+		{
+			Name: "should return an error if the client throws an error",
+			Client: errorClient,
+			EntitySource: testEntitySource,
+			ExpectedError: fmt.Errorf("something bad happened"),
+		}, 
+	} {
+		t.Run(tt.Name, func(t *testing.T)  {
+			client :=  tt.Client
+			entitySource := tt.EntitySource 
+			resolver := resolution.NewOperatorResolver(client, entitySource)
+			solution, err := resolver.Resolve(context.Background())	
+
+			if err != nil {
+				assert.Equal(t, err, tt.ExpectedError)
+				assert.Nil(t, solution)
+			} else {
+				assert.Len(t, solution.SelectedVariables(), tt.SelectedVariableCnt)
+				if len(solution.SelectedVariables()) == 4 {
+					assert.True(t, solution.IsSelected("operatorhub/packageA/2.0.0"))
+					assert.True(t, solution.IsSelected("operatorhub/prometheus/0.47.0"))
+					assert.True(t, solution.IsSelected("required package packageA"))
+					assert.True(t, solution.IsSelected("required package prometheus"))
+					assert.False(t, solution.IsSelected("operatorhub/prometheus/0.37.0"))
+				}
```

Comment:
What happens if we add another test case where we expect 3 variables to be selected? Or 4, but different 4?

Reply:
Sure I will do that. This would cover an already covered portion of the code that tests if packages described by the available Operator resources are resolved.